lib/service.js

```diff
@@ -360,13 +396,17 @@ ServicePrivate.prototype.listen = function() {
  */
 ServicePrivate.prototype.broadcast = function(type, data) {
   debug('broadcast', type, data);
+  this._broadcast('broadcast', {
+    type: type,
+    data: data
+  });
+};
+
+ServicePrivate.prototype._broadcast = function(messageId, data) {
```

Comment:
As mentioned above: If we passed the `this.channels[client]` into the `ServiceStream` it could send messages directly to the client on it's own.